### PR TITLE
mobile: turning up release validation CI

### PR DESCRIPTION
**Pull request by alyssawilk**
copied from the mobile repo, and edited to run for repo merge.  Only runs for mobile PRs.

Risk Level: low
Testing: first push shows it worked successfully
Docs Changes: n/a
Release Notes: n/a
part of https://github.com/envoyproxy/envoy/issues/23758

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -20,6 +20,7 @@
 !\.yapfignore
 !\.zuul
 !\.zuul.yaml
+!\.github/workflows/*
```
Comment (phlax) [inline .gitignore:23]: please dont do this - rather clean up the yaml

**Review by phlax — COMMENTED** [review]
looks good, altho im a little concerned that we are adding a lot of CI that will still run when not strictly required

the `paths` (and `paths-ignore`) pattern for workflow triggers is cool - i checked re azp to do the same thing - it seems we can altho i think we probably still need to use the pattern we have discussed as it is more about not/triggering steps _within_ a workflow there

i was a little bit confused by the `.gitignore` change - mostly because i could see that the yaml should not pass linting so assumed that it was to prevent linting 

i realise now that it does the opposite - ie ensure those files are included - but its not necessary as there is already:

```gitignore
 15 !\.github
```

checking this confirms

```console
$ touch .github/workflows/asdf
$ git status .github/workflows/
On branch tooling-yamllint2
Untracked files:
  (use "git add <file>..." to include in what will be committed)
	.github/workflows/asdf

```


re linting the reason these files were not picked up is due to it only searching for `.yaml` files and not `.yml` - im fixing that now (https://github.com/envoyproxy/pytooling/pull/623)

and will lint the existing files as that lands

**Comment by phlax** [comment]
yaml lint/cleanup pr is #24197

File: .gitignore (hunk 1 above)
Comment (alyssawilk) [reply .gitignore:23]: it's not necessary but it makes it a lot less annoying to change workflow files as it means git add * will add them.

File: .github/workflows/release_validation.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+name: release_validation
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+    - 'mobile/**'
```
Comment (phlax) [inline .github/workflows/release_validation.yml:8]: could you add the `.azure-pipelines/**` path also pleaes

Comment (phlax) [reply .github/workflows/release_validation.yml:8]: sorry - not azp - the `.github/workflows/*` path

**Review by jpsim — APPROVED** [review]

**Pull request merged**